node/flatpak-node-generator.py

```diff
@@ -1317,6 +1384,31 @@ def process_lockfile(self, lockfile: Path) -> Iterator[Package]:
             yield self.parse_package_section(lockfile, section)
 
 
+class YarnRCFileProvider(RCFileProvider):
+    rcfile_name = '.yarnrc'
+
+    @classmethod
+    def parse_rcfile(cls, rcfile: Path) -> Dict[str, str]:
+        parser_re = re.compile(r'^(\S+)\s+(?:"(.+)"|(\S+))', re.MULTILINE)
+        with open(rcfile, 'r') as r:
+            rcfile_text = r.read()
+        result = {}
+        for key, quoted_val, val in parser_re.findall(rcfile_text):
+            result[key] = quoted_val + val
+        print(result)
+        return result
+
+    def get_node_headers(self, rcfile: Path) -> NodeHeaders:
+        yarnrc = self.parse_rcfile(rcfile)
+        if 'target' not in yarnrc:
+            return None
+        nh_args = {'target': yarnrc['target']}
```

Comment:
I think this would be a bit nicer if it just assigned to some local variables that default to None (e.g. just `runtime = yarnrc.get('runtime')`) and passed them manually, it'll make it more resilient later on since mypy can actually check to make sure all the args match up.